### PR TITLE
Resolve cppcheck errors

### PR DESCRIPTION
**Pull request by twometresteve**
## Goal

Resolve errors found by `cppcheck` locally, also ensuring that CI fails for errors in the future.

## Changeset

I've made the changes that seem appropriate, I don't think they should have any functional bearing.

## Testing

Covered by a full CI run.

### CODE REVIEW

File: bugsnag-plugin-android-ndk/src/main/jni/bugsnag_ndk.c [hunk 1]
```diff
@@ -159,6 +159,7 @@ JNIEXPORT void JNICALL Java_com_bugsnag_android_ndk_NativeBridge_install(
   // copy event path to env struct
   const char *event_path = bsg_safe_get_string_utf_chars(env, _event_path);
   if (event_path == NULL) {
+    free(bugsnag_env);
```
Comment (kstenerud) [inline bugsnag-plugin-android-ndk/src/main/jni/bugsnag_ndk.c:162]: For cleanup code in C, it's generally safer to use a "poor man's finally" approach using goto. That way, any future checked code that can fail doesn't require you to hunt for and match up every freeable resource and every exit point in the function, and all resource-freeing code exists in one place.

```c
{
    resource *resource_a = NULL;
    resource *resource_b = NULL;
    object *some_object = NULL;

    resource_a = get_resource_a();
    if (resource_a == NULL) {
        goto exit;
    }

    if (!validate_resource(resource_a)) {
        goto exit;
    }

    resource_b = get_resource_b(resource_a);
    if (resource_b == NULL) {
        goto exit;
    }

    if (!validate_resource(resource_b)) {
        goto exit;
    }

    some_object = copy_object_from_resource(resource_b);

exit:
    free(resource_a);
    free(resource_b);
    return some_object;
}
```

File: bugsnag-plugin-android-ndk/src/main/jni/event.c [hunk 2]
```diff
@@ -119,6 +119,8 @@ bsg_metadata_value bugsnag_get_metadata_value(void *event_ptr,
   }
   bsg_metadata_value data;
   data.type = BSG_METADATA_NONE_VALUE;
+  data.bool_value = false;
+  data.double_value = 0.0;
```
Comment (kstenerud) [inline bugsnag-plugin-android-ndk/src/main/jni/event.c:123]:
```c
  bsg_metadata_value data = {
          .type = BSG_METADATA_NONE_VALUE
  };
```

**Review by kstenerud — CHANGES_REQUESTED** [review]

File: bugsnag-plugin-android-ndk/src/main/jni/bugsnag_ndk.c (hunk 1 above)
Comment (twometresteve) [reply bugsnag-plugin-android-ndk/src/main/jni/bugsnag_ndk.c]: Refactored to `free` when the allocated struct is not assigned to `bsg_global_env`.

File: bugsnag-plugin-android-ndk/src/main/jni/event.c (hunk 2 above)
Comment (twometresteve) [reply bugsnag-plugin-android-ndk/src/main/jni/event.c]: Changed as suggested.

**Review by kstenerud — APPROVED** [review]

**Pull request merged**